### PR TITLE
Fixes unnecessary legacy scheme dry import for gc admission test regression

### LINKED ISSUE #70192: Unexpectedly broken test in OwnerReferencesPermissionEnforcement admission controller
/kind flake
/sig testing
/sig auth

we are injecting a restMapper into this admisson controller while in its test, the restMapper somehow become empty when we remove internal clientset/informer dependency from it (#70167). now we come up with a walkaround in which we manually init the legacy schema via a dry import so that the built-in resources can be installed. this failing test suggests that the test is not quite correct logically.

### PR DESCRIPTION
**Pull request by yue9944882**
/kind flake
/sig api-machinery

use fake discovery instead of legacy scheme to build rest mapper

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #70192

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by deads2k** [comment]
much better.  nice job.

/lgtm

**Comment by caesarxuchao** [comment]
@deads2k it needs your approval.

"lgtm" doesn't approve anymore.

**Comment by deads2k** [comment]
/approve



> @deads2k it needs your approval.
> 
> "lgtm" doesn't approve anymore.

:(

File: plugin/pkg/admission/gc/gc_admission_test.go [hunk 1]
```diff
@@ -101,7 +101,30 @@ func newGCPermissionsEnforcement() (*gcPermissionsEnforcement, error) {
 	}
 
 	genericPluginInitializer := initializer.New(nil, nil, fakeAuthorizer{}, nil)
-	pluginInitializer := kubeadmission.NewPluginInitializer(nil, testrestmapper.TestOnlyStaticRESTMapper(legacyscheme.Scheme), nil)
+	fakeDiscoveryClient := &fakediscovery.FakeDiscovery{Fake: &coretesting.Fake{}}
+	fakeDiscoveryClient.Resources = []*metav1.APIResourceList{
+		{
+			GroupVersion: corev1.SchemeGroupVersion.String(),
+			APIResources: []metav1.APIResource{
+				{Name: "nodes", Namespaced: false, Kind: "Node"},
```
Comment (yue9944882) [inline plugin/pkg/admission/gc/gc_admission_test.go]: added v1.nodes resource add fake discovery to sync up w/ the latest change in this file

**Pull request merged**